git-ftp

```diff
@@ -1831,7 +1831,7 @@ do
 		*)
 			# Pass thru anything that may be meant for fetch.
 			if [ -n "$1" ]; then
-				if [ -z "$URL" ]; then
+				if [ -z "$URL" ] && ! echo "$1" | egrep -q '^(-[A-Za-z])|^(--[A-Za-z0-9-]+)'; then
```

Comment:
I personally prefer grep's `-v` option instead of the `!`. Do you mind?

What do we do if someone registers the domain name `-my-domain.example`? I guess they have to give the full URL with the protocol then: `ftp://-my-domain.example`

And What if someone types the option wrong? Example: `git ftp push +insecure example.net`
It would still take `+insecure` as URL, right?

Reply:
> I personally prefer grep's `-v` option instead of the `!`. Do you mind?

Oh yes. sure. I’ll have a look.

> What do we do if someone registers the domain name `-my-domain.example`? I guess they have to give the full URL with the protocol then: `ftp://-my-domain.example`

This is matched by the expression as described here: https://github.com/git-ftp/git-ftp/issues/522#issuecomment-504765313
We only get a problem if there are 2 dashes `--` as the first characters. 

> And What if someone types the option wrong? Example: `git ftp push +insecure example.net`
> It would still take `+insecure` as URL, right?

Yes you’re right. But at it will show this as “Cant connect to '+insecure'” so the user can get where the problem is, can’t he.
If you want to have some details why I choose this solution please refer to the comments for the issue #522.

Reply:
Oh no. It does not say Can’t connect but "fatal: Remote host not set.". This is because of `set_remote_host` but I can fix this by checking it there.